packages/pg/lib/client.js

```diff
@@ -718,11 +796,16 @@ class Client extends EventEmitter {
   }
 
   connect(callback) {
-    if (!this.connectionParameters.loadBalance) {
+    if (this.connectionParameters.loadBalance === 'false') {
       logger.silly("Loadbalance is false, falling to upstream behaviour")
       return this.nowConnect(callback)
     }
     lock.acquire().then(() => {
```

Comment:
We are holding the lock until the connection is created. That may be an overkill and we should release it just before we create the actual connection (`nowConnect()` may be).

Reply:
ToDo note added.